File: go/escalations.go
package aegis

import (
	"context"
	"fmt"

	"github.com/ai-aegis-yatav/aegis-sdk/go/internal"
	"github.com/ai-aegis-yatav/aegis-sdk/go/models"
)

type EscalationsService struct {
	t *internal.Transport
}

func (s *EscalationsService) Create(ctx context.Context, req models.EscalationCreateRequest) (*models.Escalation, error) {
	var resp models.Escalation
	if err := s.t.Do(ctx, "POST", "/escalations", req, &resp); err != nil {
		return nil, err
	}
	return &resp, nil
}

func (s *EscalationsService) Get(ctx context.Context, id string) (*models.Escalation, error) {
	var resp models.Escalation
	if err := s.t.Do(ctx, "GET", fmt.Sprintf("/escalations/%s", id), nil, &resp); err != nil {
		return nil, err
	}
	return &resp, nil
}

func (s *EscalationsService) List(ctx context.Context, params models.ListParams) (*models.PaginatedResponse[models.Escalation], error) {
	var resp models.PaginatedResponse[models.Escalation]
	path := buildPath("/escalations", params.ToQuery())
	if err := s.t.Do(ctx, "GET", path, nil, &resp); err != nil {
		return nil, err
	}
	return &resp, nil
}

func (s *EscalationsService) Resolve(ctx context.Context, id string, req models.EscalationResolveRequest) (*models.Escalation, error) {
	var resp models.Escalation
	if err := s.t.Do(ctx, "POST", fmt.Sprintf("/escalations/%s/resolve", id), req, &resp); err != nil {
		return nil, err
	}
	return &resp, nil
}

func (s *EscalationsService) Assign(ctx context.Context, id string, req models.EscalationAssignRequest) (*models.Escalation, error) {
	var resp models.Escalation
	if err := s.t.Do(ctx, "POST", fmt.Sprintf("/escalations/%s/assign", id), req, &resp); err != nil {
		return nil, err
	}
	return &resp, nil
}

func (s *EscalationsService) Claim(ctx context.Context, id string) (*models.Escalation, error) {
	var resp models.Escalation
	if err := s.t.Do(ctx, "POST", fmt.Sprintf("/escalations/%s/claim", id), nil, &resp); err != nil {
		return nil, err
	}
	return &resp, nil
}

func (s *EscalationsService) Stats(ctx context.Context) (*models.EscalationStats, error) {
	var resp models.EscalationStats
	if err := s.t.Do(ctx, "GET", "/escalations/stats", nil, &resp); err != nil {
		return nil, err
	}
	return &resp, nil
}

File: rust/src/resources/escalations.rs
use std::sync::Arc;

use reqwest::Method;

use crate::error::Result;
use crate::models::escalations::*;
use crate::transport::Transport;

pub struct EscalationsResource {
    transport: Arc<Transport>,
}

impl EscalationsResource {
    pub(crate) fn new(transport: Arc<Transport>) -> Self {
        Self { transport }
    }

    pub async fn create(&self, request: &EscalationCreateRequest) -> Result<Escalation> {
        self.transport
            .request(Method::POST, "/v1/escalations", Some(request))
            .await
    }

    pub async fn get(&self, id: &str) -> Result<Escalation> {
        self.transport
            .request(
                Method::GET,
                &format!("/v1/escalations/{}", id),
                None::<&()>,
            )
            .await
    }

    pub async fn list(&self, params: &EscalationListParams) -> Result<EscalationListResponse> {
        let mut query_parts = Vec::new();
        if let Some(ref status) = params.status {
            query_parts.push(format!("status={}", status));
        }
        if let Some(page) = params.page {
            query_parts.push(format!("page={}", page));
        }
        if let Some(limit) = params.limit {
            query_parts.push(format!("limit={}", limit));
        }
        let path = if query_parts.is_empty() {
            "/v1/escalations".to_string()
        } else {
            format!("/v1/escalations?{}", query_parts.join("&"))
        };
        self.transport
            .request(Method::GET, &path, None::<&()>)
            .await
    }

    pub async fn resolve(
        &self,
        id: &str,
        request: &EscalationResolveRequest,
    ) -> Result<Escalation> {
        self.transport
            .request(
                Method::POST,
                &format!("/v1/escalations/{}/resolve", id),
                Some(request),
            )
            .await
    }

    pub async fn assign(&self, id: &str, request: &EscalationAssignRequest) -> Result<Escalation> {
        self.transport
            .request(
                Method::POST,
                &format!("/v1/escalations/{}/assign", id),
                Some(request),
            )
            .await
    }

    pub async fn claim(&self, id: &str) -> Result<Escalation> {
        self.transport
            .request(
                Method::POST,
                &format!("/v1/escalations/{}/claim", id),
                None::<&()>,
            )
            .await
    }

    pub async fn stats(&self) -> Result<EscalationStats> {
        self.transport
            .request(Method::GET, "/v1/escalations/stats", None::<&()>)
            .await
    }
}
